### DISCUSSION TITLE
Welcome to go-snmp-olt-zte-c320 Discussions!

### DISCUSSION THREAD

**Discussion by sumitroajiprabowo** [Cepat-Kilat-Teknologi/go-snmp-olt-zte-c320#3] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by cherlow** [comment]
Do you have more  OIDs for  instance for onu uptime and onu distance. Is there a resource where they are documented?

**Reply by sumitroajiprabowo** [reply]
You can search for it using snmp trap or using the mib browser application to search for the oid value you want..

**Reply by cherlow** [reply]
I did use MIB Browser for most of the stuff I got. A few of them OID's don't seem to be documented anywhere. For the titan series I realised that OID's occuring at   .1.3.6.1.4.1.3902.1082.500.10. and  .1.3.6.1.4.1.3902.1082.500.20.  even in oid info sites such as http://oid-info.com  don't exist.  Do you have the MIB's ? They would be of good help. Thankyou

**Reply by sumitroajiprabowo** [reply]
You can check [this](https://www.scribd.com/document/486022802/ZXA10-C3xx-MIB-Specifications)

**Comment by cherlow** [comment]
![Screenshot from 2024-08-14 15-37-50](https://github.com/user-attachments/assets/767706f3-8473-4568-adfb-c151aae72239)
Thank you. I got this information from ZTE OLT Titan series and Huawei OLT SmartAX.

**Reply by sumitroajiprabowo** [reply]
are you using olt zte c3xx?

**Reply by cherlow** [reply]
No. zte olt c6xx but seems to be somehow compatible with c3xx from the OIDs you had.

**Reply by sumitroajiprabowo** [reply]
in the documentation that I read, I just saw that there is an ois to show last downtime, last uptime, last down cause and distance as you sent in the image. can you show the OID? Is it compatible with the C3xx OLT that I use? Thanks

**Reply by MuhammadAlfiFauzul** [reply]
> ![Screenshot from 2024-08-14 15-37-50](https://private-user-images.githubusercontent.com/28813883/357814477-767706f3-8473-4568-adfb-c151aae72239.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMxNzI2OTIsIm5iZiI6MTc1MzE3MjM5MiwicGF0aCI6Ii8yODgxMzg4My8zNTc4MTQ0NzctNzY3NzA2ZjMtODQ3My00NTY4LWFkZmItYzE1MWFhZTcyMjM5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzIyVDA4MTk1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJkMjFkMmNjMjUzNjZkMjlmM2EzZGE5NDMxZDkzNmFmYzdhODMxYzQ1ZTQ3NzM4MWI4MDk0NzM5OTEwZTA4NmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.bpeIk-gkRYFx0u9M1SOuthyjeWxhc62M_FiJ-CAOGKY) Thank you. I got this information from ZTE OLT Titan series and Huawei OLT SmartAX.

sir what oid tx power and rx power?

**Comment by cherlow** [comment]
I don't have a zte olt c3xx but you can try

- {name: gpon.ont.LastOnlineTime, oid: .1.3.6.1.4.1.3902.1082.500.10.2.3.8.1.5}

 - {name: gpon.ont.LastOfflineTime, oid: .1.3.6.1.4.1.3902.1082.500.10.2.3.8.1.6}

 - {name: gpon.ont.LastOfflineReason, oid: .1.3.6.1.4.1.3902.1082.500.10.2.3.8.1.7, values: { 1: unknown, 2: LOS, 3: LOSi, 4: LOFi, 5: sfi, 6: loai, 7: loami, 8: AuthFail, 9: PowerOff, 10: deactiveSucc, 11: deactiveFail, 12: Reboot, 13: Shutdown }}
 
- {name: gpon.optical.distance, oid: .1.3.6.1.4.1.3902.1082.500.10.2.3.10.1.2}

The lastoffline and lastonline time are octet strings that can be converted to datetime

**Reply by sumitroajiprabowo** [reply]
thanks its work for OLT ZTE c3xx

gpon ont last online time
<img width="825" alt="Screenshot 2024-08-15 at 01 40 08" src="https://github.com/user-attachments/assets/23d2c9ca-6e62-4a01-9e6c-469a2e54f003">

gpon ont last offline time
<img width="825" alt="Screenshot 2024-08-15 at 01 37 16" src="https://github.com/user-attachments/assets/b80382bb-d2bf-4eed-aaab-00d8ce993a08">


gpon ont Last offline reason
<img width="689" alt="Screenshot 2024-08-15 at 01 26 20" src="https://github.com/user-attachments/assets/f07bb886-a01b-43b6-8695-2f0c92a05ea5">

gpon optical distance
<img width="684" alt="Screenshot 2024-08-15 at 01 23 49" src="https://github.com/user-attachments/assets/6939e02f-f49e-44d2-9860-385146ebba21">

Thanks

**Comment by Dieg0CLima** [comment]
Hello, I am working in a monitoring platform and i am searching for oid bandwidth information to extract from the OLT ZTE (bytes and packets),

**Reply by sumitroajiprabowo** [reply]
sorry i'm late in replying to your message my friend, for bandwidth monitoring I dont graph from olt using snmp, I usually use nms application to do network monitoring, or use grafana which is graphed from my router.

**Reply by Dieg0CLima** [reply]
Thats Ok, Thanks for reply.

I found some doc about this, now i am working to implement performance improvements
![WhatsApp Image 2025-02-12 at 10 40 22](https://github.com/user-attachments/assets/6ef766b5-e30e-4b6f-89df-51033b25e037)

work pretty well but i want to do using **SNMP Traps**, i am have some issues to find MIBs and Doc about it, I would be very grateful if you have any information about.

This print is from the system i am building, i am creating one ERP to consolidate in one place NMS, MAP, Billing, Contract, etc.

**Reply by sumitroajiprabowo** [reply]
Wow. in your picture shows the bandwidth usage?
i think you are using snmp trap..
would you like to share oid for display bandwidth usage?

**Reply by sumitroajiprabowo** [reply]
i see in the picture you sent there are some buttons like add onu, and delete onu.

are you using snmp write to do that?

**Reply by Dieg0CLima** [reply]
The actions on this screen are automated using templates, basically telnet commands.

SNMP trap that I will use to obtain the event when my ONU has signal loss, I will receive this event from the OLT via SNMP Trap and record it in my relational database and inform my neutral network partners or register a maintenance service order, in these screens I create a JavaScript function that makes a request to my server (every 5 seconds, i make like this because it is simples, later i will change for Websockets ) that will use SNMP to obtain bandwidth usage, for example. The SNMP Trap in my case is for the source of the data to be issued from the OLT and not my server requesting SNMP data

The oids i am using:

zxAnPonOnuIfRxOctetRate	1.3.6.1.4.1.3902.1082.500.4.2.2.2.1.3
zxAnPonOnuIfTxOctetRate	1.3.6.1.4.1.3902.1082.500.4.2.2.2.1.46

**Reply by Dieg0CLima** [reply]
My difficulty with SNMP Trap is that I receive it from the OLT, but I don't have the MIBs to translate the information, I searched on ZTE and Zabbix blogs and had no success. Can you help me with this?

**Comment by suciptonetwork** [comment]
cara mengkoneksikan snmp dari olt ke snmp go bagaimana om

**Reply by sumitroajiprabowo** [reply]
bisa cek disini om [README.md](https://github.com/Cepat-Kilat-Teknologi/go-snmp-olt-zte-c320/blob/main/README.md)
selamat mencoba omku. 👨‍💻👩‍💻

**Comment by dimmascool** [comment]
Anyone have oid's for getting tx-power?

**Reply by Dieg0CLima** [reply]
.1.3.6.1.4.1.3902.1082.500.4.2.1.2.1.44 (TX Octects)

.1.3.6.1.4.1.3902.1082.500.4.2.1.2.1.45 (TX Packets)

.1.3.6.1.4.1.3902.1082.500.20.2.2.2.1.14 (TX Power)

**Reply by dimmascool** [reply]
![image](https://github.com/user-attachments/assets/84b7cb08-0980-4576-8fea-42b6a28339aa)

why the value between snmp and telnet is different, any idea ?

**Reply by Dieg0CLima** [reply]
Thats OIDs is for C320, try these instead:

.1.3.6.1.4.1.3902.1082.30.40.2.4.1.3

**Reply by msramdan** [reply]
> Itu OID untuk C320, coba yang ini sebagai gantinya:
> 
> .1.3.6.1.4.1.3902.1082.30.40.2.4.1.3

not working bro

**Comment by dimmascool** [comment]
![image](https://github.com/user-attachments/assets/b5254064-b8e7-4c7f-8d10-e472864edaa9)

Does anyone know how to get the OID index, such as "gpon-olt_1/1/1," from an unconfigured ONT, like in the image?

**Reply by sumitroajiprabowo** [reply]
![WhatsApp Image 2025-03-10 at 00 10 58](https://github.com/user-attachments/assets/c5e91dd3-b00e-403e-8270-9fbee9ed18a7)

do you mean like this?

**Reply by dimmascool** [reply]
Yup, btw i solved my problem by reading oids in your code "Monitoring OLT ZTE C320 with SNMP", thx a lot bro. Sorry my English is bad kwwkkw

**Reply by dimmascool** [reply]
Oh iya, untuk melihat onu yang belum ter registrasi apakah sudah ada di project anda bang ? saya lihat belum ada oid yang menunjukan data onu ter registrasi

**Reply by sumitroajiprabowo** [reply]
Kalo untuk unauthenticate onu saya pake interactive ssh bang.
Untuk projectnya udah ada masih tahap development

**Comment by infiplusonesolo** [comment]
does anyone know what is the zte SNMP OID to get mac addresses, like when using command 

show mac 

on ssh

here are some output example, using the ssh command 

80f7.a6ef.17ad   219   Dynamic   gpon-onu_1/2/1:2         vport 1
80f7.a6ef.1831   219   Dynamic   gpon-onu_1/2/1:3         vport 1
80f7.a6f7.c1c9   219   Dynamic   gpon-onu_1/2/1:4         vport 1
80f7.a6ef.1765   219   Dynamic   gpon-onu_1/2/1:5         vport 1

**Reply by sumitroajiprabowo** [reply]
Get mac address per pon? 🤔

**Comment by achyar10** [comment]
Hallo, kalau untuk register apa sudah ada?

**Reply by sumitroajiprabowo** [reply]
register gimana kak?

**Reply by achyar10** [reply]
untuk aktivasi, authenticate ontnya..

**Comment by seisoci** [comment]
ada info untuk oid C320 V2.1 RX & TX Power yang yml tidak terbaca

**Reply by sumitroajiprabowo** [reply]
modul ini saya menggunakan C320 Version V2.1.0

**Comment by msramdan** [comment]
<img width="1583" height="658" alt="image" src="https://github.com/user-attachments/assets/28bb51ca-d8bc-4b23-8928-da5a522efcf2" />
untuk dapat redaman ONU gmana yah,, saya itu pakai scping telnet show pon power onu-rx gpon-olt_{olt}\n sangat berat kli ada yg tau klo via SNMP gmana

**Reply by Ilhamcrazylags** [reply]
saya tau om oid sama rumus nya tapi kasi tau juga untuk onu index nya gpon-onu_1/1/1:1 kita berbagi

**Reply by sumitroajiprabowo** [reply]
> <img alt="image" width="1583" height="658" src="https://private-user-images.githubusercontent.com/67257764/507535784-28bb51ca-d8bc-4b23-8928-da5a522efcf2.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjM1MTk1OTYsIm5iZiI6MTc2MzUxOTI5NiwicGF0aCI6Ii82NzI1Nzc2NC81MDc1MzU3ODQtMjhiYjUxY2EtZDhiYy00YjIzLTg5MjgtZGE1YTUyMmVmY2YyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTE5VDAyMjgxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkxNWU1MDM3N2RlYjcwZjdhMmRiZjFmNjk0YTE2OTgwNjNiOTY0M2IwNWUzYTg2MGQwODhlN2FiZmU3NWJlOTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.dKOjLUOpnyvA0WGyDuuLIDqzeLd-XFaBCqDwm84TV6g"> untuk dapat redaman ONU gmana yah,, saya itu pakai scping telnet show pon power onu-rx gpon-olt_{olt}\n sangat berat kli ada yg tau klo via SNMP gmana

ini pake OLT apa ya om ?

**Reply by msramdan** [reply]
> saya tau om oid sama rumus nya tapi kasi tau juga untuk onu index nya gpon-onu_1/1/1:1 kita berbagi

bisa wa aja mas 083874731480

**Reply by msramdan** [reply]
> > <img alt="image" width="1583" height="658" src="https://private-user-images.githubusercontent.com/67257764/507535784-28bb51ca-d8bc-4b23-8928-da5a522efcf2.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjM1MTk1OTYsIm5iZiI6MTc2MzUxOTI5NiwicGF0aCI6Ii82NzI1Nzc2NC81MDc1MzU3ODQtMjhiYjUxY2EtZDhiYy00YjIzLTg5MjgtZGE1YTUyMmVmY2YyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTE5VDAyMjgxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkxNWU1MDM3N2RlYjcwZjdhMmRiZjFmNjk0YTE2OTgwNjNiOTY0M2IwNWUzYTg2MGQwODhlN2FiZmU3NWJlOTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.dKOjLUOpnyvA0WGyDuuLIDqzeLd-XFaBCqDwm84TV6g"> untuk dapat redaman ONU gmana yah,, saya itu pakai scping telnet show pon power onu-rx gpon-olt_{olt}\n sangat berat kli ada yg tau klo via SNMP gmana
> 
> ini pake OLT apa ya om ?

OLT zte C300 sama C320

**Reply by sumitroajiprabowo** [reply]
> > > <img alt="image" width="1583" height="658" src="https://private-user-images.githubusercontent.com/67257764/507535784-28bb51ca-d8bc-4b23-8928-da5a522efcf2.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjM1MTk1OTYsIm5iZiI6MTc2MzUxOTI5NiwicGF0aCI6Ii82NzI1Nzc2NC81MDc1MzU3ODQtMjhiYjUxY2EtZDhiYy00YjIzLTg5MjgtZGE1YTUyMmVmY2YyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTE5VDAyMjgxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkxNWU1MDM3N2RlYjcwZjdhMmRiZjFmNjk0YTE2OTgwNjNiOTY0M2IwNWUzYTg2MGQwODhlN2FiZmU3NWJlOTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.dKOjLUOpnyvA0WGyDuuLIDqzeLd-XFaBCqDwm84TV6g"> untuk dapat redaman ONU gmana yah,, saya itu pakai scping telnet show pon power onu-rx gpon-olt_{olt}\n sangat berat kli ada yg tau klo via SNMP gmana
> > 
> > 
> > ini pake OLT apa ya om ?
> 
> OLT zte C300 sama C320

modul ini di develop dari olt C320 Version V2.1.0

**Reply by nifqi18** [reply]
> <img alt="image" width="1583" height="658" src="https://private-user-images.githubusercontent.com/67257764/507535784-28bb51ca-d8bc-4b23-8928-da5a522efcf2.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njk1ODQwNTksIm5iZiI6MTc2OTU4Mzc1OSwicGF0aCI6Ii82NzI1Nzc2NC81MDc1MzU3ODQtMjhiYjUxY2EtZDhiYy00YjIzLTg5MjgtZGE1YTUyMmVmY2YyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTI4VDA3MDIzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU2MWRiYmQzYmJlOGRhM2FhMzg0OGJhNjU5YjQ3MjkxYTlkNzQxMGM3NDg3OWUxYjIwNjMxMjFiNDYwNzQ0ZWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ZvHqFW6kXuTU_WjIGCu88V9hcDATJF3WoU-ZAhB_Py8"> untuk dapat redaman ONU gmana yah,, saya itu pakai scping telnet show pon power onu-rx gpon-olt_{olt}\n sangat berat kli ada yg tau klo via SNMP gmana

Paling enak gunakan snmp agar tidak ringan , dan jangan lupa caching nya

**Comment by MuhammadAlfiFauzul** [comment]
dokumentasi oid c300 apakah ada om?

**Reply by sumitroajiprabowo** [reply]
> dokumentasi oid c300 apakah ada om?

saat ini saya hanya memiliki c320 aja om hehe.

**Comment by saharamitracom** [comment]
ada yang pernah coba create wan ip dari snmp ? oidnya pake apa ya

**Reply by kroto69** [reply]
Sepertinya snmp zte hanya support read jadi tidak bisa, mnding via telnet saja 

> ada yang pernah coba create wan ip dari snmp ? oidnya pake apa ya

**Reply by saharamitracom** [reply]
sudah bisa write dan read

**Comment by sunurich** [comment]
Hello! I have 30 ZTE OLTs — is it possible to use this tool with multiple SNMP hosts, or does it support only one OLT per instance?

**Reply by sumitroajiprabowo** [reply]
Hi @sunurich — promised update: **multi-OLT shipped in [v3.2.0](https://github.com/Cepat-Kilat-Teknologi/snmp-olt-zte/releases/tag/v3.2.0)** 🎉

Your 30 OLTs can now run in **a single instance**. Define them once (any mix of C320/C300):

```bash
# inline env, or better for 30 devices: a file you can mount as a k8s Secret
OLTS_FILE=/etc/olt/olts.json
DEFAULT_OLT=site-01
```

```json
[
  {"id":"site-01","host":"10.0.0.1","port":161,"community":"public","boards":"1,2"},
  {"id":"site-02","host":"10.0.0.2","port":161,"community":"public","boards":"3:16,5:8"}
]
```

What you get per OLT, fully isolated inside one process:

- **API dimension**: `GET /api/v1/olt/{id}/board/{slot}/pon/{pon}/...` (the `DEFAULT_OLT` still answers the bare `/api/v1/board/...` paths, so nothing breaks)
- **Own SNMP pool + namespaced Redis cache** (`olt_<id>_…`) — 30 OLTs can share one Redis with zero collisions
- **Per-OLT readiness probes**: `/readyz` reports `snmp_<id>` for each device; one unreachable OLT degrades instead of taking the pod down
- **Per-slot board topology** (`boards: "3:16,5:8"`) — C300 GTGH/GTGO mixes and C320 both work; OID generation/validation is per card
- Optional **per-tenant API keys** (`API_USERS`): each key only sees the OLTs it owns — handy if the 30 OLTs belong to different customers
- Bonus: **uplink/card auto-detect** (`GET /api/v1/olt/{id}/uplinks`) via ENTITY-MIB + IF-MIB

Helm: chart `3.2.1` adds `olt.oltsFile` (renders the registry into a Secret mounted at `/etc/olt/olts.json`) — so it's **one release for all 30**, not 30 releases:

```bash
helm repo add snmp-olt https://cepat-kilat-teknologi.github.io/snmp-olt-zte/
helm repo update
helm install olt-monitor snmp-olt/snmp-olt-zte --version 3.2.1 -f my-olts-values.yaml
```

Heads-up: the repo was renamed `go-snmp-olt-zte-c320` → **`snmp-olt-zte`** for this release (old URLs redirect). Full details in the [README](https://github.com/Cepat-Kilat-Teknologi/snmp-olt-zte#multiple-olts-in-one-instance-olts) and the [CHANGELOG](https://github.com/Cepat-Kilat-Teknologi/snmp-olt-zte/blob/main/CHANGELOG.md).

Would love to hear how it goes with your 30-OLT fleet! 🚀

**Reply by sunurich** [reply]
Wow! This is absolutely awesome! 🚀

I'm actually coming back to this project because it's a million times better than all of my own custom scripts—and honestly, better than anything else I've found for monitoring ZTE OLTs.

Seeing that multi-OLT support has been implemented is amazing. This is exactly what I was hoping for when I first asked about it. Being able to manage dozens of OLTs from a single instance is a huge improvement for real ISP environments.

Thank you to everyone who contributed to this project. Your work saves network engineers and system administrators an incredible amount of time, and it's clear that this wasn't built just as a demo—it was built by people who understand real operational needs.

I'm really looking forward to deploying it in our production environment and building even more tools on top of it.

Keep up the fantastic work! I sincerely wish this project continued growth, more contributors, and wider adoption. The ZTE community truly needed something like this.

From the bottom of my heart, thank you for making this open source. ❤️

**Comment by sumitroajiprabowo** [comment]
Hi @sunurich, thanks for the question!

**Current state:** the tool is designed as **one instance per OLT** — `SNMP_HOST` points to a single device. For your 30 OLTs, the recommended approach today is to deploy 30 Helm releases with different `snmp.host` values:

```bash
helm install snmp-olt-site-01 ./snmp-olt-zte-c320 --set snmp.host=10.0.0.1
helm install snmp-olt-site-02 ./snmp-olt-zte-c320 --set snmp.host=10.0.0.2
```

Each instance gets its own Redis cache, trap listener, and webhook config — fully isolated.

**Roadmap:** I'm planning to extend this module to support **multi-OLT management** for both **ZTE C300 and C320** in a single instance. The OID trees of C300 and C320 are largely compatible, so a unified OLT profile abstraction should work well.

Will share progress here once development starts. Stay tuned! 🚀
